tests/integration/test_tls.py

```diff
@@ -164,7 +172,93 @@ async def test_kafka_tls(ops_test: OpsTest, app_charm):
     assert private_key_2 == new_private_key
 
 
-# TODO: Add mTLS tests
+@pytest.mark.abort_on_fail
+async def test_mtls(ops_test: OpsTest):
+    # creating the signed external cert on the unit
+    action = await ops_test.model.units.get(f"{DUMMY_NAME}/0").run_action("create-certificate")
+    response = await action.wait()
+    client_certificate = response.results["client-certificate"]
+    client_ca = response.results["client-ca"]
+
+    encoded_client_certificate = base64.b64encode(client_certificate.encode("utf-8")).decode(
+        "utf-8"
+    )
+    encoded_client_ca = base64.b64encode(client_ca.encode("utf-8")).decode("utf-8")
+
+    # deploying mtls operator with certs
+    tls_config = {
+        "generate-self-signed-certificates": "false",
+        "certificate": encoded_client_certificate,
+        "ca-certificate": encoded_client_ca,
+    }
+    await ops_test.model.deploy(
+        CERTS_NAME, channel="stable", config=tls_config, series="jammy", application_name=MTLS_NAME
+    )
+    await ops_test.model.wait_for_idle(apps=[MTLS_NAME], timeout=1000, idle_period=15)
+    await ops_test.model.add_relation(
+        f"{APP_NAME}:{TRUSTED_CERTIFICATE_RELATION}", f"{MTLS_NAME}:{TLS_RELATION}"
+    )
+
+    # ensuring kafka broker restarts with new listeners
+    async with ops_test.fast_forward(fast_interval="30s"):
+        await asyncio.sleep(180)
```

Comment:
We don't have this in vm, I would hope that it's not needed on k8s since k8s is usually faster

Reply:
It's to do with something related to nodeports, hence the difference. I think deploying with nodeports slowed down 'something' (but still only in CI). The general fixes here were to attempt to address that.
The issue you highlighted here was to do with the Kafka broker not 'enabling' mtls fast enough, hence the wait.